napari/components/_interaction_box_mouse_bindings.py

```diff
@@ -168,7 +186,7 @@ def _on_transform_change(self, event):
         )
         self._ref_viewer().layers.selection.active.affine = (
             self._ref_viewer().layers.selection.active.affine.replace_slice(
-                layer_dims_displayed, event.value
+                layer_dims_displayed, event.value.compose(self._ini_transf_inv)
```

Comment:
for readability here I would separate this out to something like

```
current_affine = self._ref_viewer().layers.selection.active.affine
layer_transform = event.value.compose(self._ini_trans.inverse)
current_affine.replace_slice(layer_dims_displayed, layer_transform)
```

what do you think?

Reply:
Unfortunately this work around do not work, because then the affine event of the selected layer do not response. The event is not connected in this case. Nonetheless I have adapted the layer_transform assignment for more readability.

Reply:
works for me!